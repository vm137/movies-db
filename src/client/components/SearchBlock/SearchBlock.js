import React, {Component} from 'react';
import enterArrow from '../../img/enter-arrow.svg';
import './SearchBlock.scss';

export default class SearchBlock extends Component {

    constructor(props) {
        super(props);
        this.state = {
            movies: [],
            searchByButtons: true
        };
    }

    componentDidMount() {

        /* eslint-disable no-undef */
        fetch("http://react-cdp-api.herokuapp.com/movies")
            .then(function (response) {
                return response.json();
            })
            .then(function (myJson) {
                console.log(myJson);
            });
    }

    changeButtonsColor() {
        this.setState({searchByButtons: !this.state.searchByButtons});
    }

    render() {

        return (
          <div className="search-block-wrapper">
            <div className="search-block">
              <p className="find-title">find your movie</p>

              <div className="input-wrapper">
                <input className="search-input" type="text" />
                <img className="img-arrow" src={enterArrow} alt="enter" />
              </div>

              <div className="search-buttons">
                <span className="search-by">search by</span>
                <button
                  className={`btns-search btn-title ${this.state.searchByButtons ? "btn-bg-red" : "btn-bg-grey"}`}
                  onClick={this.changeButtonsColor.bind(this)}
                  type="button">title</button>
                <button
                  className={`btns-search btn-genre ${!this.state.searchByButtons ? "btn-bg-red" : "btn-bg-grey"}`}
                  onClick={this.changeButtonsColor.bind(this)}
                  type="button">genre</button>

                <button className="btns-search btn-search" type="submit">search</button>
              </div>

            </div>
          </div>
        )
    }

}
