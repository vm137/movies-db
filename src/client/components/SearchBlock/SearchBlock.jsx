import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchResults from '../SearchResults';
import Logo from '../Logo';
import Utils from '../Utils/Utils';
import COLOR from '../constants/constants';
import enterArrow from '../../img/enter-arrow.svg';
import './style.scss';

export default class SearchBlock extends Component {
  static propTypes = {
    numberFoundMovies: PropTypes.number.isRequired,
    movies: PropTypes.arrayOf(PropTypes.object).isRequired,
    searchCB: PropTypes.PropTypes.func.isRequired,
    onClick: PropTypes.PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      firstMount: true,
      titleButtonSelected: true,
      inputValue: 'you', // TODO: make '' after testing.
      sortByRelease: true,
    };
    this.textInput = React.createRef();
    this.onKeyPress = this.handleKeyPress.bind(this);
    this.makeSearch = this.makeSearch.bind(this);
    this.swapSearchByButtonsColor = this.swapSearchByButtonsColor.bind(this);
    this.handleSortByClick = this.handleSortByClick.bind(this);
  }


  componentDidMount() {
    this.textInput.focus();

    // TODO: delete after testing.
    this.makeSearch();
  }

  swapSearchByButtonsColor() {
    this.setState(prevState => ({ titleButtonSelected: !prevState.titleButtonSelected }));
  }

  makeSearch() {
    this.setState({ firstMount: false });
    const { titleButtonSelected, inputValue } = this.state;
    const { searchCB } = this.props;
    const searchBy = titleButtonSelected ? 'title' : 'genres';
    searchCB(inputValue, searchBy);
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value,
    });
  }

  handleSortByClick() {
    const { sortByRelease } = this.state;
    this.setState({
      sortByRelease: !sortByRelease,
    });
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      this.makeSearch();
    }
  }

  render() {
    const { numberFoundMovies, movies, onClick } = this.props;
    const {
      inputValue, firstMount, titleButtonSelected, sortByRelease,
    } = this.state;
    const sortReleaseColor = sortByRelease ? COLOR.NETFLIX_RED : COLOR.NETFLIX_GREY;
    const sortRatingColor = sortByRelease ? COLOR.NETFLIX_GREY : COLOR.NETFLIX_RED;

    return (
      <div className="search-block-wrapper">
        <div className="search-box-wrapper">
          <Logo />
          <div className="search-block">
            <p className="find-title">find your movie</p>

            <div className="input-wrapper">
              <input
                value={inputValue}
                onChange={evt => this.updateInputValue(evt)}
                onKeyPress={this.onKeyPress}
                ref={(input) => { this.textInput = input; }}
                className="search-input"
                type="text"
              />
              <img className="img-arrow" src={enterArrow} alt="enter" />
            </div>

            <div className="search-buttons">
              <span className="search-by">search by</span>
              <button
                className={`btns-search btn-title ${titleButtonSelected ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.swapSearchByButtonsColor}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
title
              </button>
              <button
                className={`btns-search btn-genre ${!titleButtonSelected ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.swapSearchByButtonsColor}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
genre
              </button>

              <button
                onClick={this.makeSearch}
                className="btns-search btn-search"
                type="submit"
              >
search
              </button>
            </div>
          </div>
        </div>
        <div className="search-details">
          <div className="number-found-movies">
            {numberFoundMovies}
            {' '}
            movies found
          </div>
          <div className="sort-by">
            Sort by:
            <span
              role="button"
              tabIndex={0}
              onKeyPress={() => {}}
              className="sort-release-date"
              style={{ color: sortReleaseColor }}
              onClick={this.handleSortByClick}
            >
              release date
            </span>
            <span
              role="button"
              tabIndex={-1}
              onKeyPress={() => {}}
              className="sort-rating"
              style={{ color: sortRatingColor }}
              onClick={this.handleSortByClick}
            >
rating
            </span>
          </div>
        </div>
        {!firstMount && (
        <SearchResults onClick={onClick} movies={movies} />
        )}
      </div>
    );
  }
}
