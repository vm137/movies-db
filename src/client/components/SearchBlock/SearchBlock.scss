.search-box-wrapper {
    height: 200px;
    position: relative;

    &::after {
        content: "";
        background: url('../../img/old-cinema.png');
        opacity: 0.8;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        position: absolute;
        z-index: -1;
    }

    .search-block {
        position: absolute;
        top: 50px;
        left: 5%;
        width: 90%;

        .find-title {
            margin-top: 12px;
            color: #fff;
            font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
            font-size: 16px;
            text-transform: uppercase;
            text-shadow: 1px 1px 2px rgb(0, 0, 0);
        }

        .input-wrapper {
            position: relative;
            margin-top: 8px;

            .search-input {
                width: 100%;
                height: 28px;
                margin-top: 12px;
                color: #fff;
                padding-left: 12px;
                padding-bottom: 4px;
                border: none;
                border-bottom: 1px #ff594b solid;
                outline: none;
                font-size: 14px;
                box-sizing: border-box;
                background-color: #1d1d1d;
            }

            .img-arrow {
                position: absolute;
                width: 20px;
                top: 16px;
                right: 8px;
            }
        }

        .search-buttons {
            margin-top: 22px;

            .search-by {
                color: #fff;
                font-size: 12px;
                font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
                text-transform: uppercase;
            }
            .btns-search {
                border: 0;
                color: #fff;
                width: 90px;
                height: 26px;
                border-radius: 4px;
                outline: none;
                font-size: 12px;
                font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
                text-transform: uppercase;

                &:hover {
                    background-color: #ff866c;
                }
                &:active {
                    background-color: #151515;
                }
            }
            .btn-title {
                margin-left: 10px;
            }
            .btn-genre {
                margin-left: 16px;
            }
            .btn-search {
                float: right;
                background-color: #ff594b;
            }
        }
    }
}

.btn-bg-red {background-color: #ff594b !important;}
.btn-bg-grey {background-color: #808080 !important;}
