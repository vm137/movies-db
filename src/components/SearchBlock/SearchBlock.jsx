import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchResults from '../SearchResults';
import Logo from '../Logo';
import Utils from '../Utils/Utils';
import COLOR from '../../constants/constants';
import enterArrow from '../../img/enter-arrow.svg';
import './style.scss';

export default class SearchBlock extends Component {
  static propTypes = {
    fetchMovies: PropTypes.func.isRequired,
    movies: PropTypes.arrayOf(Object).isRequired,
    totalR: PropTypes.number.isRequired,
    swapSearchBy: PropTypes.func.isRequired,
    swapSortBy: PropTypes.func.isRequired,
    searchBy: PropTypes.bool.isRequired,
    sortBy: PropTypes.bool.isRequired,
    history: PropTypes.objectOf(PropTypes.any),
    match: PropTypes.shape({
      params: PropTypes.shape({
        query: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    history: {},
    match: {
      params: {
        query: '',
      },
    },
  };

  constructor(props) {
    super(props);
    this.state = {
      inputValue: '',
    };
    this.textInput = React.createRef();
    this.onKeyPress = this.handleKeyPress.bind(this);
    this.handleSearchBy = this.handleSearchBy.bind(this);
    this.handleSortBy = this.handleSortBy.bind(this);
    this.handleSearchClick = this.handleSearchClick.bind(this);
  }

  componentDidMount() {
    const { fetchMovies, match, searchBy } = this.props;
    const { query } = match.params;
    const searchByParam = searchBy ? 'title' : 'genres';
    if (query && query.length > 0) {
      this.setState({
        inputValue: query,
      });
      fetchMovies(query, searchByParam);
    }

    this.textInput.focus();
  }

  handleSearchClick() {
    const { inputValue } = this.state;
    const { fetchMovies, searchBy, history } = this.props;
    const searchByParam = searchBy ? 'title' : 'genres';

    if (inputValue.length > 0) {
      history.push(`/search/${inputValue}`);
      fetchMovies(inputValue, searchByParam);
    }
  }

  handleSearchBy() {
    const { swapSearchBy } = this.props;
    swapSearchBy();
  }

  handleSortBy() {
    const { swapSortBy, movies, sortBy } = this.props;
    swapSortBy(movies, sortBy);
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value, // TODO: make search hints
    });
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      this.handleSearchClick();
    }
  }

  render() {
    const { totalR, searchBy, sortBy } = this.props;
    const { inputValue } = this.state;
    const sortReleaseColor = sortBy ? COLOR.NETFLIX_RED : COLOR.NETFLIX_GREY;
    const sortRatingColor = sortBy ? COLOR.NETFLIX_GREY : COLOR.NETFLIX_RED;

    return (
      <div className="search-block-wrapper">
        <div className="search-box-wrapper">
          <Logo />
          <div className="search-block">
            <p className="find-title">find your movie</p>

            <div className="input-wrapper">
              <input
                value={inputValue}
                onChange={evt => this.updateInputValue(evt)}
                onKeyPress={this.onKeyPress}
                ref={(input) => { this.textInput = input; }}
                className="search-input"
                type="text"
              />
              <img className="img-arrow" src={enterArrow} alt="enter" />
            </div>

            <div className="search-buttons">
              <span className="search-by">search by</span>
              <button
                className={`btns-search btn-title ${searchBy ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.handleSearchBy}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
title
              </button>
              <button
                className={`btns-search btn-genre ${!searchBy ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.handleSearchBy}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
genre
              </button>

              <button
                onClick={this.handleSearchClick}
                className="btns-search btn-search"
                type="submit"
              >
search
              </button>
            </div>
          </div>
        </div>
        <div className="search-details">
          <div className="number-found-movies">
            {totalR}
            {' '}
            movies found
          </div>
          <div className="sort-by">
            Sort by:
            <span
              role="button"
              tabIndex={0}
              onKeyPress={() => {}}
              className="sort-release-date"
              style={{ color: sortReleaseColor }}
              onClick={this.handleSortBy}
            >
              release date
            </span>
            <span
              role="button"
              tabIndex={-1}
              onKeyPress={() => {}}
              className="sort-rating"
              style={{ color: sortRatingColor }}
              onClick={this.handleSortBy}
            >
rating
            </span>
          </div>
        </div>
        <SearchResults />
      </div>
    );
  }
}
