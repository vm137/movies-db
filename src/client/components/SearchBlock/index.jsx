import { connect } from 'react-redux';
import SearchBlock from './SearchBlock';
import { fetchMoviesAction } from '../../actions/actions';

const mapStateToProps = state => ({
  moviesTotalR: state.total,
});

const mapDispatchProps = dispatch => ({
  fetchMovies: (searchString, searchBy) => dispatch(fetchMoviesAction(searchString, searchBy)),
});

export default connect(mapStateToProps, mapDispatchProps)(SearchBlock);
