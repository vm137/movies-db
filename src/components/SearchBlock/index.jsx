import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import SearchBlock from './SearchBlock';
import { fetchMoviesAction, swapSearchByAction, swapSortByAction } from '../../actions/actions';

const mapStateToProps = state => ({
  totalR: state.total,
  movies: state.movies,
  searchBy: state.searchBy,
  sortBy: state.sortBy,
});

const mapDispatchProps = dispatch => ({
  fetchMovies: (searchString, searchBy) => dispatch(fetchMoviesAction(searchString, searchBy)),
  swapSearchBy: () => dispatch(swapSearchByAction()),
  swapSortBy: (movies, sortBy) => dispatch(swapSortByAction(movies, sortBy)),
});

export default withRouter(connect(mapStateToProps, mapDispatchProps)(SearchBlock));
