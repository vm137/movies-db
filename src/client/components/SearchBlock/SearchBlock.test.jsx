import { mount } from 'enzyme';
import React from 'react';
import SearchBlock from './SearchBlock';

const moviesTotalR = 10;
const fetchMovies = () => {};

describe('<SearchBlock />', () => {
  it.skip('renders matching snapshot', () => {
    const wrapper = mount(<SearchBlock
      moviesTotalR={moviesTotalR}
      fetchMovies={fetchMovies}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('check for .search-block-wrapper', () => {
    const wrapper = mount(<SearchBlock
      moviesTotalR={moviesTotalR}
      fetchMovies={fetchMovies}
    />);
    expect(wrapper.find('.search-block-wrapper').length).toBe(1);
  });

  it.skip('check for state change after click SortBy', () => {
    const wrapper = mount(<SearchBlock
      moviesTotalR={moviesTotalR}
      fetchMovies={fetchMovies}
    />);

    const button = wrapper.find('.sort-release-date');
    expect(button.length).toBe(1);

    let status = wrapper.state().sortByRelease;
    expect(status).toEqual(true);
    button.simulate('click');
    status = wrapper.state().sortByRelease;
    expect(status).toEqual(false);
  });
});
