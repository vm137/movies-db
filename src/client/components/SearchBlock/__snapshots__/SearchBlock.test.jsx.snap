// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBlock /> renders matching snapshot 1`] = `
<SearchBlock
  movies={Array []}
  numberFoundMovies={10}
  onClick={[Function]}
  searchCB={[Function]}
>
  <div
    className="search-block-wrapper"
  >
    <div
      className="search-box-wrapper"
    >
      <Logo>
        <p
          className="logo"
        >
          netflixroulette
        </p>
      </Logo>
      <div
        className="search-block"
      >
        <p
          className="find-title"
        >
          find your movie
        </p>
        <div
          className="input-wrapper"
        >
          <input
            className="search-input"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />
          <img
            alt="enter"
            className="img-arrow"
            src="test-file-stub"
          />
        </div>
        <div
          className="search-buttons"
        >
          <span
            className="search-by"
          >
            search by
          </span>
          <button
            className="btns-search btn-title btn-bg-red"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            title
          </button>
          <button
            className="btns-search btn-genre btn-bg-grey"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            genre
          </button>
          <button
            className="btns-search btn-search"
            onClick={[Function]}
            type="submit"
          >
            search
          </button>
        </div>
      </div>
    </div>
    <div
      className="search-details"
    >
      <div
        className="number-found-movies"
      >
        10
         
        movies found
      </div>
      <div
        className="sort-by"
      >
        Sort by:
        <span
          className="sort-release-date"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          style={
            Object {
              "color": "#ff594b",
            }
          }
          tabIndex={0}
        >
          release date
        </span>
        <span
          className="sort-rating"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          style={
            Object {
              "color": "#464646",
            }
          }
          tabIndex={-1}
        >
          rating
        </span>
      </div>
    </div>
  </div>
</SearchBlock>
`;
